pkg/controller/seed-controller-manager/kubernetes/cluster_controller.go

```diff
@@ -322,7 +322,12 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, clus
 		return nil, fmt.Errorf("failed to sync health: %w", err)
 	}
 
-	res, err := r.reconcileCluster(ctx, cluster, namespace)
+	conf, err := r.configGetter(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get KubermaticConfiguration: %w", err)
+	}
```

Comment:
Whenever a feature gate in the KubermaticConfiguration changes, the KKP operator will adjust the seed/master-controller-manager Deployments and set the `--feature-gates` flag accordingly.

So during the runtime of the controller manager, this flag cannot change (well, it can, but briefly after that the Pod will be terminated and replaced with a new one).

Similar to what Rafraf did in #14425, you can and should just take the feature gates that are loaded in the main and hand them into this controller.

Reply:
thanks for the details! i've updated the code in 1e4cf07d5adc689a5e219be7ce8c4c5c73b31f9c